compose.yml

```diff
@@ -0,0 +1,79 @@
+version: '3' 
+
+services:
+
+  bitcoind:
+    image: kylemanna/bitcoind:latest
+    volumes:
+      - data:/bitcoin/.bitcoin
+    ports: 
+      - "8332:8332"                                        # mainnet
+      - "18332:18332"                                      # testnet
+    command: 
+      - "-chain=${COUNTERPARTY_NETWORK:-main}"             # export COUNTERPARTY_NETWORK=test for testnet
+      - "-rpcallowip=0.0.0.0/0"
+      - "-rpcbind=0.0.0.0"
+      - "-rpcuser=rpc"
+      - "-rpcpassword=rpc"
+      - "-listen=1"
+      - "-server=1"
+      - "-printtoconsole=1"
+      - "-addresstype=legacy"
+      - "-txindex=1"
+      - "-prune=0"
+      - "-dbcache=4000"
+      - "-mempoolfullrbf=1"
+
+
+  addrindexrs:
+    image: counterparty/addrindexrs:v0.4.6
+    links:
+      - bitcoind
+    volumes:
+      - data:/root/.bitcoin
+      - data:/data
+    ports:
+      - "8432:8432"                                        # mainnet
+      - "18432:18432"                                      # testnet
+    command:
+      - "--network=${COUNTERPARTY_NETWORK:-main}"          # export COUNTERPARTY_NETWORK=test for testnet
+      - "--indexer-rpc-host=0.0.0.0"
+      - "--daemon-rpc-host=bitcoind"
+      - "--cookie=rpc:rpc"
+      - "-vvv"
+      - "--db-dir=/data/"
+    environment:
+      - ADDRINDEXRS_JSONRPC_IMPORT=${ADDRINDEXRS_JSONRPC_IMPORT:-false}
+
+
+  counterparty-core:
+    image: counterparty/counterparty:v10.0.0-rc.1
+    links:
+      - bitcoind
+      - addrindexrs
+    volumes:
+      - data:/root/.bitcoin
+      - data:/data
+    ports:
+      - "4000:4000"                                        # mainnet
+      - "14000:14000"                                      # testnet
+    command:
+      - start
+      - "--${COUNTERPARTY_NETWORK:-main}net"               # export COUNTERPARTY_NETWORK=test for testnet
+      - "--backend-connect=bitcoind"
+      - "--indexd-connect=addrindexrs"
+      - "--rpc-host=0.0.0.0"
+      - "--catch-up=bootstrap"
+      #- "--verbose"
+    environment:
+      - "XDG_DATA_HOME=/data/"
+      - "XDG_LOG_HOME=/data/"
+
+
+volumes: 
+  data:
+    driver: local
+    driver_opts:
+      type: none
+      o: bind
+      device: ${COUNTERPARTY_DOCKER_DATA:-~/counterparty-docker-data}/
```

Comment:
Hm, I think we should put this somewhere else by default... maybe `$XDG_CACHE_HOME/counterparty-docker-data` I don't want to pollute people's home dirs.

Reply:
hm.. $XDG_CACHE_HOME is in people's home dir.. by default it's `~/.cache`.. and the counterparty data is not really cache..
what do you think of `${COUNTERPARTY_DOCKER_DATA:-~/.local/share/counterparty-docker-data}`

Reply:
hm yeah we could use `$XDG_DATA_HOME`, which is ~/.local/share` by default

(I was thinking "cache" only because the data *can* be reproduced, but `DATA` is probably better)

Reply:
> yep, we could use $XDG_DATA_HOME, which is ~/.local/share` by default!

`XDG_DATA_HOME` is not set most of the time! and docker-compose will not replace `XDG_DATA_HOME` by the default folder but by an empty string..
we could have something like this:

`${COUNTERPARTY_DOCKER_DATA:-${XDG_DATA_HOME:-~/.local/share}}/counterparty-docker-data`

(in this cas data will be always in `counterparty-docker-data` subfolder)

Reply:
that works!